app/Resources/assets/js/jquery.instantSearch.js

```diff
@@ -0,0 +1,79 @@
+// jQuery plugin for an instant searching
+(function($) {
+    $.fn.instantSearch = function(config) {
+        return this.each(function() {
+            initInstantSearch(this, $.extend(true, defaultConfig, config || {}));
+        });
+    };
+
+    var defaultConfig = {
+        minQueryLength: 2,
+        maxPreviewItems: 10,
+        previewDelay: 500,
+        noItemsFoundMessage: 'No items found'
+    };
+
+    var throttle = (function(){
+        var timer = 0;
+        return function(callback, ms){
+            clearTimeout (timer);
+            timer = setTimeout(callback, ms);
+        };
+    })();
+
+    var initInstantSearch = function(el, config) {
+        var $input = $(el);
+        var $form = $input.parents('form').first();
```

Comment:
use ``$form = $input.closest('form')`` instead. It will be more efficient as it will stop traversing the parents when it finds the first form instead of searching them all to keep only the first one.